File: src/main/java/com/travel/global/util/MessageUtil.java
package com.travel.global.util;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.MessageSource;
import org.springframework.stereotype.Component;

import java.util.Locale;

@Component
public class MessageUtil {
    private static MessageSource messageSource;

    @Autowired
    public void setMessageSource(MessageSource messageSource) {
        MessageUtil.messageSource = messageSource;
    }

    public static String getMessage(String message) {
        try {
            return messageSource.getMessage(message, null, Locale.KOREA);
        } catch (Exception e) {
            return message + " not found message";
        }
    }
}

File: src/main/kotlin/com/travel/global/util/MessageUtil.kt
package com.travel.global.util

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.MessageSource
import org.springframework.stereotype.Component
import java.util.*


@Component
class MessageUtil {
    @Autowired
    fun setMessageSource(messageSource: MessageSource?) {
        Companion.messageSource = messageSource
    }

    companion object {
        private var messageSource: MessageSource? = null
        fun getMessage(message: String): String {
            return try {
                messageSource!!.getMessage(message, null, Locale.KOREA)
            } catch (e: Exception) {
                "$message not found message"
            }
        }
    }
}
